newsroom/news_api/app/__init__.py

```diff
@@ -103,16 +107,122 @@ def base_exception_error(err):
 def get_app(config=None):
     app = NewsroomNewsAPI(__name__, config=config)
 
+    def convert_to_syndicate(data, token, formatter):
+        # remove token paramerters from requirments
+        if formatter and formatter == 'ATOM':
+            return NewsAPISyndicateService.generate_atom_feed(data)
+        elif formatter and formatter == 'RSS':
+            return NewsAPISyndicateService.generate_rss_feed(data)
+        elif formatter and formatter == 'JSON':
+            return jsonify(data)
+        else:
+            raise ValueError("Invalid formatter specified")
+
+    def handle_unsupported_format(data, token, formatter):
+        error_message = f"Unsupported formatter: {formatter}"
+        error_response = make_response(jsonify({'error': error_message}), 400)
+        return process_error_response(error_response)
+
+    FORMAT_HANDLERS = defaultdict(
+        lambda: {'handler': handle_unsupported_format, 'content_type': 'application/json'},
+        {
+            'ATOM': {'handler': convert_to_syndicate, 'content_type': 'application/xml'},
+            'RSS': {'handler': convert_to_syndicate, 'content_type': 'application/xml'},
+            'JSON': {'handler': convert_to_syndicate, 'content_type': 'application/json'},
+        }
+    )
+
     @app.after_request
-    def after_request(response):
+    def process_response(response):
+        if 'news/syndicate' in request.url:
```

Comment:
Any `syndicate` specific code would be best placed under the `newsroom.news_api.news.syndicate` module, and not in the top level `newsroom.news_api.app` module.

This can be achieved in the `init_app` method, using Flasks `after_request` registration function